proton-c/src/core/transport.c

```diff
@@ -1648,7 +1659,7 @@ int pn_do_disposition(pn_transport_t *transport, uint8_t frame_type, uint16_t ch
   bool remote_data = (pn_data_next(transport->disp_data) &&
                       pn_data_get_list(transport->disp_data) > 0);
 
-  for (pn_sequence_t id = first; id <= last; id++) {
+  for (pn_sequence_t id = first; sequence_less_than(id, last) || (id == last); id++) {
```

Comment:
Of course this test will still do the wrong thing if the wrap-around is big enough (ie if first > last at teh start of the loop)